play-json/src/test/scala/play/api/libs/json/MacroSpec.scala

```diff
@@ -63,11 +63,54 @@ class MacroSpec extends org.specs2.mutable.Specification {
         }
       }
     }
+
+    "be generated for a sealed family" >> {
+      implicit val simpleReads = Reads[Simple] { js =>
+        (js \ "bar").validate[String].map(Simple(_))
+      }
+      implicit val optionalReads: Reads[Optional] = Json.reads[Optional]
+      implicit val familyReads: Reads[Family] = Json.reads[Family]
+
+      val simple = Simple("foo")
+      val optional = Optional(None)
+
+      "using the _value syntax" in {
+        val jsSimple = Json.obj(
+          "_type" -> "play.api.libs.json.MacroSpec.Simple",
+          "_value" -> Json.writes[Simple].writes(simple)
+        )
+
+        val jsOptional = Json.obj(
+          "_type" -> "play.api.libs.json.MacroSpec.Optional",
+          "_value" -> Json.writes[Optional].writes(optional)
+        )
+
+        jsSimple.validate[Family].get must beTypedEqualTo(simple) and {
```

Comment:
Yes, as the `Reads` should anyway check for `_value` in case of non-`JsObject`, there is no extra cost to do it also for `JsObject`